Content.Client/Paint/PaintVisualizerSystem.cs

```diff
@@ -0,0 +1,105 @@
+using System.Linq;
+using Robust.Client.GameObjects;
+using static Robust.Client.GameObjects.SpriteComponent;
+using Content.Shared.Clothing;
+using Content.Shared.Hands;
+using Content.Shared.Paint;
+
+namespace Content.Client.Paint
+{
+    public sealed class PaintedVisualizerSystem : VisualizerSystem<PaintedComponent>
+    {
+        /// <summary>
+        /// Visualizer for Paint which applies a shader and colors the entity.
+        /// </summary>
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<PaintedComponent, HeldVisualsUpdatedEvent>(OnHeldVisualsUpdated);
+            SubscribeLocalEvent<PaintedComponent, ComponentShutdown>(OnShutdown);
+            SubscribeLocalEvent<PaintedComponent, EquipmentVisualsUpdatedEvent>(OnEquipmentVisualsUpdated);
+        }
+
+        // Applies the shader and color to all sprite layers for the entity.
+        protected override void OnAppearanceChange(EntityUid uid, PaintedComponent component, ref AppearanceChangeEvent args)
+        {
+            if (args.Sprite == null)
+                return;
+
+            if (!TryComp(uid, out SpriteComponent? sprite))
+                return;
+
+            for (var layer = 0; layer < args.Sprite.AllLayers.Count(); ++layer)
+            {
+                sprite.LayerSetColor(layer, component.Color);
+                sprite.LayerSetShader(layer, component.ShaderName);
```

Comment:
It'd probably be a good idea for the shader to only be set if there isn't already a shader. As a bonus, this makes it possible to easily remove the paint without potential side effects (as you'd only need to remove the shader if it's the paint shader)

Reply:
any idea how to check if it has a shader on a layer?, this does not effect a postshader such as stealth.

Reply:
In checking the files only 2 entities currently use layershaders so it is easier to include them in a component exclusion.

Reply:
Just iterate AllLayers instead of indexing because layers aren't an array and there's no guarantee int keys exist.

Reply:
Does this mean using a foreach as opposed to for loop?